tests/unit_tests/test_pychop.py

```diff
@@ -288,6 +321,9 @@ def test_nonfermi_detector_width_squared(e_init, pychop_nonfermi_data):
 
 
 def _test_get_detector_width_squared(e_init, cls, data, pychop):
+    if data.detector is None:
+        pytest.skip('This instrument does not have a detector defined')
```

Comment:
Could this be inferred from the Class? Why would we have objects with an empty detector attribute?

Reply:
Don't ask me why the LET instrument has no information about the detector _or_ the sample in PyChop, but it doesn't. I have it on the list of things to ask Duc. In either case, in the current implementation of PyChop, the detector and sample contributions to the resolution are optional, and depend on whether the yaml files contain the relevant parameters.

Reply:
I see! There should be a cleaner way to do this test, but it can be part of a more general overhaul of how the test cases are constructed.